src/browserlib/extract-dfns.mjs

```diff
@@ -225,6 +228,311 @@ export default function (spec, idToHeading = {}) {
     .map(node => definitionMapper(node, idToHeading));
 }
 
+function preProcessEcmascript() {
+  const sectionFilter= ":not([legacy])";
+
+  const wrapWithDfn = (el) => {
+    // wrap with a dfn
+    const dfn = document.createElement("dfn");
+    for (let child of [...el.childNodes]) {
+      dfn.appendChild(child);
+    }
+    el.appendChild(dfn);
+    // set id
+    dfn.setAttribute("id", el.parentNode.getAttribute("id"));
+    dfn.dataset.ltNodefault = true;
+    return dfn;
+  };
+
+  const cleanMethodName = (name) => {
+    return name.replace(/\[/g, '')
+      .replace(/\]/g, '') // removing brackets used to mark optional args
+      .replace(/ \( */, '(')
+      .replace(/ *\)/, ')')
+      .replace(/ *,/g, ','); // trimming internal spaces
+  };
+
+  let definitionNames = new Set();
+  let idlTypes = {};
+
+  // We find the list of abstract methods
+  // to help with scoping abstract operations
+  let abstractMethods = {};
+  const abstractMethodCaptions = [...document.querySelectorAll("figcaption")]
+        .filter(el => el.textContent.match(/(abstract|additional) method/i) && el.parentNode.querySelector("emu-xref"));
+  for (let figcaption of abstractMethodCaptions) {
+    const scope = figcaption.querySelector("emu-xref").textContent;
+    const table = figcaption.parentNode.querySelector("tbody");
+    for (let td of table.querySelectorAll("tr td:first-child")) {
+      // We only consider the name of the method, not the potential parameters
+      // as they're not necessarily consistently names across
+      // the list and the definition
+      const methodName = td.textContent.trim().split('(')[0];
+      abstractMethods[methodName] = scope;
+    }
+  }
+
+  const sectionNumberRegExp = /^([A-Z]\.)?[0-9\.]+ /;
+  [...document.querySelectorAll(`${sectionFilter} h1`)].
+    forEach(el => {
```

Comment:
In theory, you don't need to create an array: 

```suggestion
  document.querySelectorAll(`${sectionFilter} h1`).forEach(el => {
```

Reply:
fixed in 6ab544f